web/ui/src/Instances/instancesFactory.js

```diff
@@ -16,7 +16,19 @@
         $notification = _notification;
         utils = _utils;
 
-        var newFactory = new BaseFactory(Instance, resourcesFactory.getRunningServices);
```

Comment:
in order to reduce the size of the running instances request, try to only get running instances for a specific service or a specific host. Otherwise, make the much larger `getRunningServices` request.